src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,299 @@
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req_vr =
+      CreateCommand<AddCommandRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(1));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg)[strings::msg_params][strings::cmd_id] = 1;
+
+  am::event_engine::Event event(hmi_apis::FunctionID::VR_AddCommand);
+  event.set_smart_object(*msg);
+
+  smart_objects::SmartObject* ptr = NULL;
+  ON_CALL(*mock_app, FindCommand(_)).WillByDefault(Return(ptr));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app, AddCommand(_, _));
+  EXPECT_CALL(*mock_app, UpdateHash());
+
+  am::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(
+          Return(DataAccessor<am::CommandsMap>(commands_map, lock_)));
+
+  req_vr->Run();
+  req_vr->on_event(event);
+}
+
+TEST_F(AddCommandRequestTest, OnEvent_BothSend_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  msg_params[strings::cmd_id] = 2;
+  msg_params[strings::vr_commands] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  msg_params[strings::vr_commands][0] = "lamer";
+
+  smart_objects::SmartObject menu_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  menu_params[strings::position] = 10;
+  menu_params[strings::menu_name] = "LG";
+
+  msg_params[strings::menu_params] = menu_params;
+  msg_params[strings::type] = 34;
+  msg_params[strings::grammar_id] = 12;
+  msg_params[strings::app_id] = 20;
+
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req_ui =
+      CreateCommand<AddCommandRequest>(msg);
+
+  MessageSharedPtr msg1 = CreateMessage(smart_objects::SmartType_Map);
+  (*msg1)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg1)[strings::msg_params][strings::cmd_id] = 1;
+  am::event_engine::Event event_ui(hmi_apis::FunctionID::UI_AddCommand);
+  event_ui.set_smart_object(*msg1);
+
+  am::event_engine::Event event_vr(hmi_apis::FunctionID::VR_AddCommand);
+  event_vr.set_smart_object(*msg1);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(1));
+
+  EXPECT_CALL(*mock_app, RemoveCommand(_)).Times(0);
+
+  req_ui->Run();
+  req_ui->on_event(event_ui);
+  req_ui->on_event(event_vr);
```

Comment:
@LevchenkoS ok you probably right.
@wolfylambova22 please add expectations for App manager as well